src/Type/grammar.pp

```diff
@@ -2,22 +2,28 @@
 
 %token parenthesis_ <
 %token _parenthesis >
+%token empty_string ""|''
```

Comment:
This looks weird, I think this should be handled as part of strings themselves, meaning that they will no longer be a single token but a node instead.

Reply:
That was the only way I found to make it work with empty strings.
Better solutions are welcome

Reply:
If you make the inner token optional in simple_type (`<quoted_string>?` and `<apostrophed_string>?`), it will work.

Reply:
Of course, you made the token optional (nullable), so the visitor needs to be updated accordingly. :)